packages/flutter/lib/src/painting/decoration_image.dart

```diff
@@ -295,6 +295,7 @@ class DecorationImagePainter {
   void _handleImage(ImageInfo value, bool synchronousCall) {
     if (_image == value)
       return;
+    _image?.image.dispose();
```

Comment:
Similarly to what's happening in rendering/image.dart, there could be an optimization here if value is a clone of _image, no?

Reply:
Adding this.